### PR TITLE
Xsoar rf 2.1

### PR DESCRIPTION
**Pull request by jonathan-s**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/content/pull/10830

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by ChanochShayner — COMMENTED** [review]
Thank you for your contribution, it looks great!

File: Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py [hunk 1]
```diff
@@ -668,70 +1149,89 @@ def main() -> None:
...
+            if params.get("offset", None):
+                params["from"] = params.pop("offset")
+            if params.get("triggered_time", None):
+                date, _ = parse_date_range(
+                    params.pop("triggered_time"),
+                    date_format="%Y-%m-%d %H:%M:%S",
+                )
+                params["triggered"] = "[{},)".format(date)
 
             return_results(get_alerts_command(client, params))
```
[the remark this answers is not in the document]
Comment (jonathan-s) [reply Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:1217]: @ChanochShayner, On line 1229 we catch any exception that occurs and return that error. Is that enough?

[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:1217]: That's fine but it's a pretty general post, it seems to be worth checking it out in each command on its own and returning a message that would be user friendly.

[the remark this answers is not in the document]
Comment (jonathan-s) [reply Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:1217]: Do you have any suggestion as how you would like to make it more user friendly?

To me the current error message is pretty clear. It tells you that the execution was unsuccessful and gives you the error output. I'm not sure what else you could ask for to be honest. 
```
        return_error(
            f"Failed to execute {demisto.command()} command. "
            f"Error: {str(e)}"
        )
```

[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:1217]: Ok, you right.

**Comment by ChanochShayner** [comment]
@jonathan-s 
https://github.com/demisto/etc/issues/30453
Are you aware of this issue?
And does your PR now have a fix for this issue?
Thanks.

**Comment by jonathan-s** [comment]
@ChanochShayner 
> @jonathan-s
> [demisto/etc#30453](https://github.com/demisto/etc/issues/30453)
> Are you aware of this issue?
> And does your PR now have a fix for this issue?
> Thanks.

Just a heads up, that link gives me a 404. So I likely don't have permission to see that.

**Comment by ChanochShayner** [comment]
> @ChanochShayner
> 
> > @jonathan-s
> > [demisto/etc#30453](https://github.com/demisto/etc/issues/30453)
> > Are you aware of this issue?
> > And does your PR now have a fix for this issue?
> > Thanks.
> 
> Just a heads up, that link gives me a 404. So I likely don't have permission to see that.

The new Recorded Future partner contribution doesn't support a suspicious threshold and has a hardcoded 5 value which leads to many false positives for legit indicators such a google, ibm, outlook, etc.
RF v1 contains this parameter.

**Comment by jonathan-s** [comment]
@ChanochShayner Fixed here -> https://github.com/demisto/content/pull/10830/files#diff-70a479bb2ddd21051961ac8828ca0b2f166da1bc512bf9de49aa21a1715c8765R212-R216

**Comment by ChanochShayner** [comment]
> @ChanochShayner Fixed here -> https://github.com/demisto/content/pull/10830/files#diff-70a479bb2ddd21051961ac8828ca0b2f166da1bc512bf9de49aa21a1715c8765R212-R216

Great!
I added the link to the issue in this PR, when it closes it will also close the issue.

**Comment by ChanochShayner** [comment]
A demo has been scheduled with @jonathan-s for Monday.

**Comment by ChanochShayner** [comment]
Demo was held today with @jonathan-s,
It was great!!
There are still small additions that need to be added, then the pr ready for merge.

File: Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py [hunk 2]
```diff
@@ -668,70 +1149,89 @@ def main() -> None:
...
+            context = demisto_args.get("context")
+            entities = {
+                "ip": argToList(demisto_args.get("ip")),
+                "domain": argToList(demisto_args.get("domain")),
+                "hash": argToList(demisto_args.get("file")),
+                "url": argToList(demisto_args.get("url")),
+                "vulnerability": argToList(demisto_args.get("cve")),
+                "filter": demisto_args.get("filter") == "yes"
+            }
             return_results(triage_command(client, entities, context))
```
[the remark this answers is not in the document]
Comment (jonathan-s) [reply Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:1192]: I addressed this in another comment further down.

File: Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py [hunk 3]
```diff
@@ -668,70 +1149,89 @@ def main() -> None:
...
-        elif command == 'recordedfuture-alert-rules':
-            rule_name = demisto_args.get('rule_name', '')
-            limit = demisto_args.get('limit', 10)
+        elif command == 'recordedfuture-single-alert':
+            _id = demisto_args['id']
+            return_results(get_alert_single_command(client, _id))
+        elif command == "recordedfuture-alert-rules":
+            rule_name = demisto_args.get("rule_name", "")
+            limit = demisto_args.get("limit", 10)
             return_results(get_alert_rules_command(client, rule_name, limit))
```
[the remark this answers is not in the document]
Comment (jonathan-s) [reply Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py:1199]: I addressed this in another comment further down.

File: Packs/RecordedFuture/IndicatorFields/indicatorfield-threatassessment.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+    "id": "indicator_threatassessment",
```
[the remark this answers is not in the document]
Comment (jonathan-s) [reply Packs/RecordedFuture/IndicatorFields/indicatorfield-threatassessment.json]: Fixed.

File: Packs/RecordedFuture/IndicatorFields/indicatorfield-riskrules.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{
+    "id": "indicator_riskrules",
+    "version": -1,
+    "modified": "2020-11-10T15:15:25.39842636Z",
+    "name": "Risk rules",
+    "ownerOnly": false,
+    "cliName": "riskrules",
```
[the remark this answers is not in the document]
Comment (jonathan-s) [reply Packs/RecordedFuture/IndicatorFields/indicatorfield-riskrules.json]: Fixed

File: Packs/RecordedFuture/IndicatorFields/indicatorfield-riskrules.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
+{
+    "id": "indicator_riskrules",
+    "version": -1,
+    "modified": "2020-11-10T15:15:25.39842636Z",
+    "name": "Risk rules",
```
[the remark this answers is not in the document]
Comment (jonathan-s) [reply Packs/RecordedFuture/IndicatorFields/indicatorfield-riskrules.json]: Fixed

File: Packs/RecordedFuture/IndicatorFields/indicatorfield-riskrules.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{
+    "id": "indicator_riskrules",
```
[the remark this answers is not in the document]
Comment (jonathan-s) [reply Packs/RecordedFuture/IndicatorFields/indicatorfield-riskrules.json]: Fixed

File: Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.yml [hunk 8]
```diff
@@ -8,9 +8,9 @@ image:
 description: Unique threat intel technology that automatically serves up relevant
   insights in real time.
 configuration:
-- display: Server URL (e.g., https://api.recordedfuture.com/v2/)
-  name: server
-  defaultvalue: https://api.recordedfuture.com/v2/
+- display: Server URL (e.g., https://api.recordedfuture.com/gw/xsoar/)
+  name: server_url
+  defaultvalue: https://api.recordedfuture.com/gw/xsoar/
```
Comment (jonathan-s) [inline Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.yml:13]: @ChanochShayner This is the new url that we set as default for the api. 

Let me know if you've got any further questions around this.

**Review by ChanochShayner — APPROVED** [review]

**Pull request merged**